pkg/controller/node/node_controller.go

```diff
@@ -101,50 +102,17 @@ func (r *ReconcileNode) Reconcile(request reconcile.Request) (reconcile.Result,
 	}
 
 	//TODO: Manage deletes
-	nodeNetworkStateKey := types.NamespacedName{
-		Namespace: "default",
-		Name:      request.Name,
-	}
-	// Create NodeNetworkState for this node
 	nodeNetworkState := &nmstatev1.NodeNetworkState{}
-	err = r.client.Get(context.TODO(), nodeNetworkStateKey, nodeNetworkState)
+	err = nmstate.GetNodeNetworkState(r.client, request.Name, nodeNetworkState)
```

Comment:
node_controller should initialize with empty state. node_network_status_controller should populate it. Let's make sure we split those responsibilities. Later, when we have Conditions implemented, we can mark it here as Initialized=True, but Populated=False. Or something similar.

Reply:
Changing currentState to empty means calling "Update" on client and the reponsable of this is node_network_status_controller, do we really need to put empty state here (In case is not already empty).

Reply:
we don't change anything here, we are creating initial NodeNetworkState. there is no NodeNetworkState to update, we trigger initial creation. the rest will be done by other component. sounds like a good distributed system approach. maybe i just can't see your concern, could you elaborate little more?

Reply:
So do we have to change something here or not ?

Reply:
- You are reconciling a created Node here
- You try to find existing NodeNetworkState on 106
- That always fails on 108, we are the ones creating it initially later in this function

We should only call CreateNodeNetworkState(), you can drop nodeNetworkState parameter, since it will be always empty while creating.

Reply:
Better logic would be to re-create the object if it's there to cover corner case, what do you think ?

Reply:
I'm fine with smarter logic, like somebody removed NodeNetworkState? We will recreate if needed. However, it has to be an empty config (in this controller).

Reply:
Done